src/bacnet/basic/object/netport.c

```diff
@@ -2851,6 +2885,88 @@ bool Network_Port_IPv6_DHCP_Server_Set(
     return status;
 }
 
+/**
+ * For a given object instance-number, gets the IPv6_DHCP_Lease_Time.
+ * Note: depends on Network_Type being set to PORT_TYPE_BIP6 for this object
+ *
+ * @param  object_instance - object-instance number of the object
+ *
+ * @return IPv6_DHCP_Lease_Time
+ */
+uint32_t Network_Port_IPv6_DHCP_Lease_Time(uint32_t object_instance)
+{
+    const unsigned index = Network_Port_Instance_To_Index(object_instance);
+    uint32_t value = 0;
+
+    if (index < BACNET_NETWORK_PORTS_MAX) {
+        if (Object_List[index].Network_Type == PORT_TYPE_BIP6) {
+            value = Object_List[index].Network.IPv6.IP_DHCP_Lease_Time;
+        }
+    }
+
+    return value;
+}
+
+/**
+ * For a given object instance-number, sets the IPv6 DHCP  lease-time
+ * Note: depends on Network_Type being set for this object
+ *
+ * @param  object_instance - object-instance number of the object
+ * @param  value - 32 bit IPv6 DHCP Lease Time
+ *
+ * @return  true if ipv6-dhcp-lease-time was set
+ */
+bool Network_Port_IPv6_DHCP_Lease_Time_Set(
+    uint32_t object_instance, const uint32_t value)
+{
+    bool status = false;
+    unsigned index = Network_Port_Instance_To_Index(object_instance);
+
+    if (index < BACNET_NETWORK_PORTS_MAX) {
+        if (Object_List[index].Network_Type == PORT_TYPE_BIP6) {
+#if (__STDC__) && (__STDC_VERSION__ >= 199901L)
+            struct timeval tv = { .tv_sec = 0, .tv_usec = 0 };
+#else
+            struct timeval tv = { 0, 0 };
+#endif
+
+            Object_List[index].Network.IPv6.IP_DHCP_Lease_Time = value;
+            if (gettimeofday(&tv, NULL) != 0) {
```

Comment:
please use mstimer for the DHCP lease time tracking in this basic object.  I don't think datetime_local() - calendar time - is the right time in seconds to use here.